File: 2022.01.24/python/15649.py
n,m = map(int,input().split())
check_arr = [False for _ in range(n+1)]

def solution(n,cnt,target,s):
  if cnt == target:
    return s+"\n"
  result = ""
  for i in range(1,n+1):
    if (not check_arr[i]):
      check_arr[i] = True
      result += solution(n,cnt+1,target,s+f'{str(i)} ')
      check_arr[i] = False
      
  return result
print(solution(n,0,m,""))

File: Boj.단계별/14_백트래킹/javascript/15649.js
const [n,m] = require('fs').readFileSync('/dev/stdin').toString().split('\n')[0].split(' ').map(Number);
const checkArr = new Array(n+1).fill(false);
const solution = (n,cnt,target,s) => {
    if (cnt === target) return s+"\n";
    let result = "";
    for (let i = 1 ; i<=n ; i++){
        if (! checkArr[i]){
            checkArr[i] = true;
            result += solution(n,cnt+1,target,s+`${i} `)
            checkArr[i] = false;
        }
    }
    return result;
};
console.log(solution(n,0,m,""));